--- origsrc/emotion-0.1.0.042/src/lib/Makefile.am	2007-11-13 00:01:32.000000000 -0600
+++ src/emotion-0.1.0.042/src/lib/Makefile.am	2008-12-25 21:17:15.521750000 -0600
@@ -18,4 +18,4 @@
 
 libemotion_la_LIBADD       = @EVAS_LIBS@ @ECORE_LIBS@
 libemotion_la_DEPENDENCIES = $(top_builddir)/config.h
-libemotion_la_LDFLAGS      = -version-info @version_info@
+libemotion_la_LDFLAGS      = -version-info @version_info@ -no-undefined
--- origsrc/emotion-0.1.0.042/src/modules/Makefile.am	2007-11-13 00:01:32.000000000 -0600
+++ src/emotion-0.1.0.042/src/modules/Makefile.am	2008-12-25 21:19:44.834250000 -0600
@@ -31,7 +31,7 @@
 emotion_xine.h \
 emotion_xine_vo_out.c
 xine_la_LIBADD       = @EVAS_LIBS@ @ECORE_LIBS@ @XINE_LIBS@ $(top_builddir)/src/lib/libemotion.la -lpthread
-xine_la_LDFLAGS      = -module -avoid-version
+xine_la_LDFLAGS      = -module -avoid-version -no-undefined
 xine_la_DEPENDENCIES = $(top_builddir)/config.h
 endif
 
@@ -42,6 +42,6 @@
 emotion_gstreamer_pipeline.c \
 emotion_gstreamer_pipeline.h
 gstreamer_la_LIBADD  = @EVAS_LIBS@ @ECORE_LIBS@ @GST_LIBS@ $(top_builddir)/src/lib/libemotion.la
-gstreamer_la_LDFLAGS = -module -avoid-version
+gstreamer_la_LDFLAGS = -module -avoid-version -no-undefined
 gstreamer_la_DEPENDENCIES = $(top_builddir)/config.h
 endif
